internal/machine/update.go

```diff
@@ -18,6 +18,76 @@ func Update(ctx context.Context, m *api.Machine, input *api.LaunchMachineInput)
 		colorize    = io.ColorScheme()
 	)
 
+	if input != nil && input.Config != nil && input.Config.Guest != nil {
+		// Check that there's a valid number of CPUs
+		var validNumCpus []int
+
+		if input.Config.Guest.CPUKind == "shared" {
+			validNumCpus = append(validNumCpus, 1, 2, 4, 6, 8)
+
+		} else if input.Config.Guest.CPUKind == "performance" {
+			validNumCpus = append(validNumCpus, 1, 2, 4, 6, 8, 10, 12, 14, 16)
+
+		}
+
+		validCpuNum := false
+
+		for _, num := range validNumCpus {
+			if num == input.Config.Guest.CPUs {
+				validCpuNum = true
+				break
+
+			}
+		}
+
+		if !validCpuNum {
+			return fmt.Errorf("invalid config: invalid number of CPUs for %s guest. Valid numbers are %v\nView more information here: https://fly.io/docs/about/pricing/#machines", input.Config.Guest.CPUKind, validNumCpus)
+
+		}
+
+		if input.Config.Guest.CPUKind == "shared" && input.Config.Guest.MemoryMB%256 != 0 {
+			return fmt.Errorf("invalid config: invalid memory size; must be in 256 MiB increment\nView more information here: https://fly.io/docs/about/pricing/#machines")
```

Comment:
how about adding a suggestion like:

```diff
-			return fmt.Errorf("invalid config: invalid memory size; must be in 256 MiB increment\nView more information here: https://fly.io/docs/about/pricing/#machines")
+			return fmt.Errorf("invalid config: invalid memory size %d; must be in 256 MiB increment (%d would work)\nView more information here: https://fly.io/docs/about/pricing/#machines", input.Config.Guest.MemoryMB, input.Config.Guest.MemoryMB - (input.Config.Guest.MemoryMB%256))
```

Reply:
That's a great idea, I'll add this in a new PR